File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.ts
function findMin(nums: number[]): number {
    let left = 0 
    let right = nums.length -1 

    if (nums[left] < nums[right]) return nums[left]

    while (right-left > 1) {
        let mid = Math.floor((left + right) / 2)

        if (nums[mid] > nums[left]) {
            left = mid
        } else {
            right = mid
        }
    }
    return nums[right]
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) -1

        if nums[left] < nums[right]:
            return nums[left]

        while left + 1 < right:
            mid = (right + left) // 2

            if nums[mid] > nums[right]:
                left = mid
            else:
                right = mid

        return nums[right]